python/lsst/analysis/tools/tasks/photometricCatalogMatch.py

```diff
@@ -0,0 +1,235 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ("PhotometricCatalogMatchConfig", "PhotometricCatalogMatchTask")
+
+
+import lsst.geom
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+from astropy.table import Table
+from astropy.time import Time
+from lsst.pipe.tasks.loadReferenceCatalog import LoadReferenceCatalogTask
+
+from ..actions.vector import VisitPlotFlagSelector
+from ..tasks.catalogMatch import CatalogMatchConfig, CatalogMatchConnections, CatalogMatchTask
+
+
+class PhotometricCatalogMatchConnections(CatalogMatchConnections):
+    pass
+
+
+class PhotometricCatalogMatchConfig(
+    CatalogMatchConfig, pipelineConnections=PhotometricCatalogMatchConnections
+):
+    filterNames = pexConfig.ListField[str](
+        doc="Physical filter names to persist downstream.",
+        default=["HSC-G", "HSC-R", "HSC-I", "HSC-Z", "HSC-Y"],
+    )
+
+    def setDefaults(self):
+        super().setDefaults()
+        self.referenceCatalogLoader.doReferenceSelection = False
+        self.referenceCatalogLoader.doApplyColorTerms = True
+
+
+class PhotometricCatalogMatchTask(CatalogMatchTask):
+    """A wrapper task to provide the information that
+    is specific to the photometric reference catalog.
+    """
+
+    ConfigClass = PhotometricCatalogMatchConfig
+    _DefaultName = "analysisToolsPhotometricCatalogMatch"
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        """Run the matching to the photometric reference
+        catalog.
+
+        Parameters
+        ----------
+        `butlerQC` : lsst.pipe.base.butlerQuantumContext.ButlerQuantumContext
+        `inputRefs` : lsst.pipe.base.connections.InputQuantizedConnection
+        `outputRefs` : lsst.pipe.base.connections.OutputQuantizedConnection
+
+        """
+
+        inputs = butlerQC.get(inputRefs)
+        bands = []
+        for filterName in self.config.filterNames:
+            bands.append(self.config.referenceCatalogLoader.refObjLoader.filterMap[filterName])
+
+        columns = self.prepColumns(bands)
+        table = inputs["catalog"].get(parameters={"columns": columns})
+        inputs["catalog"] = table
+
+        tract = butlerQC.quantum.dataId["tract"]
+
+        loaderTask = LoadReferenceCatalogTask(
+            config=self.config.referenceCatalogLoader,
+            dataIds=[ref.dataId for ref in inputRefs.refCat],
+            name=inputs["refCat"][0].ref.datasetType.name,
+            refCats=inputs["refCat"],
+        )
+
+        skymap = inputs.pop("skymap")
+        loadedRefCat = self._loadRefCat(loaderTask, skymap[tract])
+        outputs = self.run(catalog=inputs["catalog"], loadedRefCat=loadedRefCat, bands=bands)
+
+        butlerQC.put(outputs, outputRefs)
+
+
+class PhotometricCatalogMatchVisitConnections(
+    CatalogMatchConnections,
```

Comment:
Same as for `AstrometricCatalogMatchVisitConnections`: I don't think you want to inherit.